File: hashmap/205-isomorphic-strings/IsomorphicStrings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        stot = {}
        ttos = {}
        for sc, tc in zip(s, t):
            if sc in stot:
                if stot[sc]!=tc:
                    return False
            else:
                stot[sc]=tc
            if tc in ttos:
                if ttos[tc]!=sc:
                    return False
            else:
                ttos[tc]=sc
        return True

File: hashmap/205-isomorphic-strings/IsomorphicStrings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length()!=t.length()){
            return false;
        }
        HashMap<Character, Character> stot = new HashMap<>();
        HashMap<Character, Character> ttos = new HashMap<>();
        for(int i=0;i<s.length();i++){
            char sc = s.charAt(i);
            char tc = t.charAt(i);
            if(stot.containsKey(sc)){
                if(stot.get(sc)!=tc){
                    return false;
                }
            }else{
                stot.put(sc,tc);
            }
            if(ttos.containsKey(tc)){
                if(ttos.get(tc)!=sc){
                    return false;
                }
            }else{
                ttos.put(tc,sc);
            }
        }
        return true;
    }
}